app/uk/gov/hmrc/leakdetection/services/ScanningService.scala

```diff
@@ -135,8 +137,12 @@ class ScanningService @Inject()(
         }
       ))
 
-  def queueRequest(p: PushUpdate): Future[Boolean] =
-    githubRequestsQueue.pushNew(p).map(_ => true)
+  def queueDistinctRequest(p: PushUpdate): Future[Boolean] = for {
+    itemAlreadyQueued <- githubRequestsQueue.findByCommitIdAndBranch(p)
```

Comment:
`itemAlreadyQueued` sounds like a Boolean, could we do `.map(_.isDefined)` on the RHS?